### PR TITLE
add YoudaoNote.app V2.5.1

### PR DESCRIPTION
**Pull request by wuliang142857**
*If there’s a checkbox you can’t complete for any reason, that's okay, just explain in detail why you weren’t able to do so.*

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Review by miccal — APPROVED** [review]

File: Casks/youdao-note.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+cask 'youdao-note' do
+  version '2.5.1'
```
Comment (vitorgalvao) [inline Casks/youdao-note.rb]: Seeing as `url` doesn’t point to a specific version, this cask should use `version :latest` and `sha256 :no_check`. [See an example](https://github.com/caskroom/homebrew-cask/blob/37f8d181d1789cbb06c6e73625587fdbe77ca225/Casks/kibako.rb#L2L3).

File: Casks/youdao-note.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 '1ea5f4ebaacf658f712d1e2cd386c6eb90c8e4a82f577b9adbc44692b020a755'
+
+  # download.ydstatic.com/notewebsite/downloads was verified as official when first introduced to the cask
+  url 'http://download.ydstatic.com/notewebsite/downloads/YoudaoNote.dmg'
+  name '有道云笔记'
```
Comment (vitorgalvao) [inline Casks/youdao-note.rb]: This also need an english name above. See [docs](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/name.md).

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by magicsky** [comment]
OK, I added english name; and the app name should be 'YoudaoNote.app', becasuse it will be a error if app name is 'Youdao Note.app':
screen_shot__2017-03-05_22 18 37

File: Casks/youdao-note.rb (hunk 1 above)
Comment (magicsky) [inline Casks/youdao-note.rb]: OK, thank you

**Review by wuliang142857 — COMMENTED** [review]
OK, that's right

File: Casks/youdao-note.rb (hunk 2 above)
Comment (magicsky) [inline Casks/youdao-note.rb]: ok. thank you, add english name now

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**